api/src/main/java/club/mondaylunch/gatos/api/controller/NodeTypesController.java

```diff
@@ -26,4 +35,17 @@ public List<NodeTypeInfo> getNodeTypes() {
             .map(NodeTypeInfo::new)
             .toList();
     }
+
+    @SuppressWarnings("unchecked")
+    public static Map<String, String> getTranslatedDisplayNameMap(String langFile) {
+        Gson gson = new Gson();
+        var file = NodeTypesController.class.getClassLoader().getResourceAsStream("display_names/" + langFile + ".json");
```

Comment:
this works fine if just one plugin provides display names, but every plugin must provide their own, and then only one file will actually be read here.